code/modules/antagonists/bloodsuckers/structures/bloodsucker_crypt.dm

```diff
@@ -0,0 +1,723 @@
+		/// This will hurt your eyes.
+		else if(held_item.tool_behaviour == TOOL_WELDER)
+			if(held_item.use_tool(src, user, 0, volume = 5))
+				torture_time -= 6
+				torture_dmg_burn += 5
+		held_item.play_tool_sound(target)
+	/// Minimum 5 seconds.
+	torture_time = max(50, torture_time * 10)
+	/// Now run process.
+	if(!do_mob(user, target, torture_time * mult))
+		return FALSE
+	/// Success?
+	if(held_item)
+		playsound(loc, held_item.hitsound, 30, 1, -1)
+		held_item.play_tool_sound(target)
+	target.visible_message(
+		span_danger("[user] performs a ritual, spilling some of [target]'s blood from their [target_string] and shaking them up!"),
+		span_userdanger("[user] performs a ritual, spilling some blood from your [target_string], shaking you up!"),
+	)
+	INVOKE_ASYNC(target, /mob.proc/emote, "scream")
+	target.Jitter(5)
+	target.apply_damages(brute = torture_dmg_brute, burn = torture_dmg_burn, def_zone = (selected_bodypart ? selected_bodypart.body_zone : null)) // take_overall_damage(6,0)
+	return TRUE
+
+/// Offer them the oppertunity to join now.
+/obj/structure/bloodsucker/vassalrack/proc/do_disloyalty(mob/living/user, mob/living/target)
+	spawn(10)
+		/// Are we still torturing? Did we cancel? Are they still here?
+		if(use_lock && target && target.client)
+			to_chat(user, span_notice("[target] has been given the opportunity for servitude. You await their decision..."))
+			var/alert_text = "You are being tortured! Do you want to give in and pledge your undying loyalty to [user]?"
+		/*	if(HAS_TRAIT(target, TRAIT_MINDSHIELD))
+				alert_text += "\n\nYou will no longer be loyal to the station!" */
+			alert_text += "\n\nYou will not lose your current objectives, but they come second to the will of your new master!"
+			to_chat(target, span_cultlarge("THE HORRIBLE PAIN! WHEN WILL IT END?!"))
+			var/list/torture_icons = list(
+				"Accept" = image(icon = 'icons/mob/actions/actions_bloodsucker.dmi', icon_state = "power_recup"),
+				"Refuse" = image(icon = 'icons/obj/items_and_weapons.dmi', icon_state = "stunbaton_active")
+				)
+			var/torture_response = show_radial_menu(target, src, torture_icons, radius = 36, require_near = TRUE)
+			switch(torture_response)
+				if("Accept")
+					disloyalty_accept(target)
+				else
+					disloyalty_refuse(target)
+	if(!do_torture(user,target, 2))
+		return FALSE
+
+	// NOTE: We only remove loyalties when we're CONVERTED!
+	return TRUE
+
+/obj/structure/bloodsucker/vassalrack/proc/RequireDisloyalty(mob/living/user, mob/living/target)
+	var/datum/antagonist/bloodsucker/bloodsuckerdatum = IS_BLOODSUCKER(user)
+	return bloodsuckerdatum.AmValidAntag(target) || HAS_TRAIT(target, TRAIT_MINDSHIELD)
+
+/obj/structure/bloodsucker/vassalrack/proc/disloyalty_accept(mob/living/target)
+	// FAILSAFE: Still on the rack?
+	if(!(locate(target) in buckled_mobs))
+		return
+	// NOTE: You can say YES after torture. It'll apply to next time.
+	disloyalty_confirm = TRUE
+	if(HAS_TRAIT(target, TRAIT_MINDSHIELD))
```

Comment:
do not like this going through mindshields

Reply:
they can say yes or no to the convert prompt and if they say no it shouldn't go through their mindshield and shouldn't convert them

Reply:
pressing yes on this prompt is literally textbook self-antagging